### ISSUE TITLE
[package] dcmtk/3.6.7: Loading builtin dicom dictionary has been disabled by default

### ISSUE THREAD

**Issue by xeospetrov** [conan-io/conan-center-index#20464]
### Description

Commit [5d8800386dc399dfd2f01cd599d4a88ff0dc9801](https://github.com/conan-io/conan-center-index/commit/5d8800386dc399dfd2f01cd599d4a88ff0dc9801) has deprecated the options `builtin_dictionary` and `external_dictionary` in favor of `default_dict` and `use_dcmdictpath`. For some reason (oversight?), the default behavior was also changed from using the builtin/external dictionary to only using the external dictionary. This is problematic for all setups that rely on the builtin dictionary, as it leads to a runtime error when initializing the `GlobalDcmDataDictionary`.

To keep the previous behavior, the correct value for `default_dict` should be `"builtin"` instead. This does not mean that the external dictionary loading doesn't work anymore, as the external loading relies on env var `DCMDICTPATH` and `use_dcmdictpath` (default `True`). Though loading through `DCM_DICT_DEFAULT_PATH` will not work anymore, but that didn't work anyway since the value was/is being malformed (another bug?).

Relevant code in `recipes/dcmtk/all/conanfile.py`:
Before:
```
"builtin_dictionary": None, # Actually means "use default". Default for DCMTK_ENABLE_BUILTIN_DICTIONARY is 1.
"external_dictionary": None, # Actually means "use default". Default for DCMTK_ENABLE_EXTERNAL_DICTIONARY is 1.
```
After:
```
"default_dict": "external",
"use_dcmdictpath": True,
"builtin_dictionary": "deprecated",
"external_dictionary": "deprecated",
```
Also take a look at `DcmDataDictionary::loadExternalDictionaries()` inside `dcmdata/libsrc/dcdict.cc`.

### Package and Environment Details

* Package Name/Version: **dcmtk/3.6.7**
* Operating System+version: **Linux Ubuntu 18.04**
* Compiler+version: **GCC 7**
* Conan version: **conan 1.58.0**
* Python version: **Python 3.7.16**

### Conan profile

[settings]
os=Linux
os_build=Linux
arch=x86_64
arch_build=x86_64
compiler=gcc
compiler.version=7
compiler.libcxx=libstdc++
build_type=Release
[options]
[conf]
[build_requires]
[env]

### Steps to reproduce

conan install dcmtk/3.6.7@ -s build_type=Debug -s compiler=gcc -s compiler.version=7 -s compiler.libcxx=libstdc++11 -g=cmake --build=dcmtk

### Logs

Not relevant.

**Comment by SpaceIm** [comment]
The default value upstream is "builtin" if Windows, "external" otherwise: https://github.com/DCMTK/dcmtk/blob/DCMTK-3.6.7/CMake/dcmtkPrepare.cmake#L212-L216

When `default_dict` is `external`, `VirtualRunEnv` provides `DCMDICTPATH` (it points to `dicom.dic` inside package), so it should work fine. If you are outside of conan generators and run your executable without VirtualRunEnv, well it's a different problem but outside of conan responsibility.

**Comment by xeospetrov** [comment]
> The default value upstream is "builtin" if Windows, "external" otherwise

Yes, these are the DCMTK defaults. Conan recipes also have defaults. The previous recipe was using the DCMTK defaults which enabled both the builtin and external (Linux). Though looking at the previous DCMTK versions that used `DCMTK_ENABLE_BUILTIN_DICTIONARY`, I am a bit puzzled. According to [dcmtkPrepare.cmake](https://github.com/DCMTK/dcmtk/blob/DCMTK-3.6.5/CMake/dcmtkPrepare.cmake#L135) the builtin dict is turned off for Linux build environments. At the same time, the [documentation](https://support.dcmtk.org/docs-snapshot/file_datadict.html) contradicts this and my local build confirms this contradiction (using 3.6.5 with previous recipe with default options). So I suspect something else is going on, but can't immediately pinpoint it. 

> When default_dict is external, VirtualRunEnv provides DCMDICTPATH

I think the `DCMDICTPATH` should be set based on the `use_dcmdictpath` option, not the `default_dict` option. This seems more logical. Then the only difference remaining between `builtin` and `external` is whether there will be a builtin dictionary being used as backup or not. This means that using `builtin` as the default value of `default_dict`  is a more safe choice, since it will not only be compatible with conan generators, but also other setups.
